### PR TITLE
Dynamic config file

### PR DESCRIPTION
**Pull request by TrevorDecker**
This will speed things up for rolls a lot.

The idea is that we now have a configuration panel on the gui that allows us to set which ports to use.  

Eventually this panel will allow for us to reconnect to sensors and change calibration values quickly.  For now it will just stop us from having to edit paths in eclipse.  I do not think that we will have reset working reliably by Saturday morning so the way we will update values is to save the config file and reopen alice.  This will cause for the config file to be read and settings to updated. 

I really want this rolling this weekend, as it will speed things up and reduce the number of mistakes we can make.

### CODE REVIEW

**Comment by TrevorDecker** [comment]
Note I do not think it is quite ready for merge, as it does not yet pass style, and still needs config save/load to be finished.  It will also need a lot of testing.

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 1]
```diff
@@ -7,10 +7,17 @@
 
 import gnu.io.CommPortIdentifier;
 
+import java.awt.Window;
+import java.io.File;
+import java.io.IOException;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.io.IOException'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l12)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 2]
```diff
@@ -7,10 +7,17 @@
 
 import gnu.io.CommPortIdentifier;
 
+import java.awt.Window;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.Window'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l10)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 3]
```diff
@@ -7,10 +7,17 @@
 
 import gnu.io.CommPortIdentifier;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'gnu.io.CommPortIdentifier'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l8)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 4]
```diff
@@ -7,10 +7,17 @@
...
 
+import java.awt.Window;
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.lang.management.ManagementFactory'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l13)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 5]
```diff
@@ -7,10 +7,17 @@
...
+import java.net.URISyntaxException;
 import java.util.ArrayList;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.ArrayList'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l15)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 6]
```diff
@@ -7,10 +7,17 @@
...
 import java.util.Enumeration;
 import java.util.List;
 
+import jdk.nashorn.internal.runtime.regexp.joni.Config;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'jdk.nashorn.internal.runtime.regexp.joni.Config'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l19)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 7]
```diff
@@ -7,10 +7,17 @@
 
 import gnu.io.CommPortIdentifier;
 
+import java.awt.Window;
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.net.URISyntaxException;
 import java.util.ArrayList;
 import java.util.Enumeration;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Enumeration'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l16)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 8]
```diff
@@ -7,10 +7,17 @@
...
 import java.util.List;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.List'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l17)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 9]
```diff
@@ -7,10 +7,17 @@
 
 import gnu.io.CommPortIdentifier;
 
+import java.awt.Window;
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.net.URISyntaxException;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.net.URISyntaxException'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l14)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java [hunk 10]
```diff
@@ -7,10 +7,17 @@
 
 import gnu.io.CommPortIdentifier;
 
+import java.awt.Window;
+import java.io.File;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.io.File'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2245811026/issues/source?bid=2938665&fileBranchId=3180106#l11)

**Comment by VivaanBahl** [comment]
I think this is looking really good and has the majority of the functionality we really want. All that's left is to overhaul the style and clean up some

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 11]
```diff
@@ -1,5 +1,19 @@
...
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.orsoncharts.util.json.JSONObject;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid unused imports such as 'com.orsoncharts.util.json.JSONObject'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l15)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 12]
```diff
@@ -1,5 +1,19 @@
...
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid unused imports such as 'com.google.gson.JsonElement'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l13)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 13]
```diff
@@ -30,20 +44,267 @@
...
+	 */
+	public static boolean isGps_enabled() {
+		return gpsEnabled;
+	}
+
+	/**
+	 * sets a new value to gpsEnabled
+	 * @param gps_enabled
+	 */
+	public static void setGps_enabled(boolean gps_enabled) {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l241)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 14]
```diff
@@ -1,5 +1,19 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid unused imports such as 'com.google.gson.GsonBuilder'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l12)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 15]
```diff
@@ -30,20 +44,267 @@
...
+	 */
+	public static String getWAYPOINT_SOURCE_LOG_FILE() {
+		return waypointSourceLogFile;
+	}
+
+	/**
+	 * sets a new value to waypointSourceLogFile
+	 * @param wAYPOINT_SOURCE_LOG_FILE
+	 */
+	public static void setWAYPOINT_SOURCE_LOG_FILE(
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l192)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 16]
```diff
@@ -30,20 +44,267 @@
...
+	 */
+	public static boolean isDataPlayBack() {
+		return DATA_PLAY_BACK;
+	}
+
+	/**
+	 * evaluates to the current value of comPortImu 
+	 * @return
+	 */
+	public static String getCOM_PORT_IMU() {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l152)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 17]
```diff
@@ -30,20 +44,267 @@
...
+	 */
+	public static String getCONFIG_FILE() {
+		return configFile;
+	}
+
+	/**
+	 * sets a new value to configFile
+	 * @param cONFIG_FILE
+	 */
+	public static void setCONFIG_FILE(String cONFIG_FILE) {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l176)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 18]
```diff
@@ -30,20 +44,267 @@
...
+	 */
+	public static void setCOM_PORT_GPS(String cOM_PORT_GPS) {
+		comPortGps = cOM_PORT_GPS;
+	}
+
+	/**
+	 * evaluates to the current value of gpsEnabled
+	 * @return
+	 */
+	public static boolean isGps_enabled() {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l233)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 19]
```diff
@@ -30,20 +44,267 @@
...
+	 */
+	public static String getPORT_VISION() {
+		return portVision;
+	}
+
+	/**
+	 * sets a new value to portVision
+	 * @param pORT_VISION
+	 */
+	public static void setPORT_VISION(String pORT_VISION) {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l289)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 20]
```diff
@@ -30,20 +44,267 @@
...
+	 */
+	public static void setCOM_PORT_IMU(String cOM_PORT_IMU) {
+		comPortImu = cOM_PORT_IMU;
+	}
+
+	/**
+	 * evaluates to the current value of configFile
+	 * @return
+	 */
+	public static String getCONFIG_FILE() {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l168)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java (hunk 19 above)
Comment (ramensandwich) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: What's up with the casing here?

Comment (VivaanBahl) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Note that most of these commits were at 4 am :D

Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: It comes from the getters and setters being auto generated when the variable names were all caps.  The variables use to be final's.  I will be going though a fixing style during class today.

**Comment by TrevorDecker** [comment]
@VivaanBahl Can you test and check this update

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java (hunk 1 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.io.IOException'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2264186395/issues/source?bid=2938665&fileBranchId=3180106#l12)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.Window'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2264186395/issues/source?bid=2938665&fileBranchId=3180106#l10)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java (hunk 3 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'gnu.io.CommPortIdentifier'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2264186395/issues/source?bid=2938665&fileBranchId=3180106#l8)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java (hunk 4 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.lang.management.ManagementFactory'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2264186395/issues/source?bid=2938665&fileBranchId=3180106#l13)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java (hunk 6 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'jdk.nashorn.internal.runtime.regexp.joni.Config'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2264186395/issues/source?bid=2938665&fileBranchId=3180106#l19)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java (hunk 8 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.List'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2264186395/issues/source?bid=2938665&fileBranchId=3180106#l17)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java (hunk 10 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyMainFile.java]: Codacy Issue found: [Avoid unused imports such as 'java.io.File'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2264186395/issues/source?bid=2938665&fileBranchId=3180106#l11)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/ConfigurationPanel.java [hunk 21]
```diff
@@ -0,0 +1,278 @@
...
+			@Override
+			public void actionPerformed(ActionEvent e) {
+		    	RobobuggyConfigFile.loadConfigFile();
+			}
+		});
+		addComponent(loadConfigButton, .8, 0, .1, .1);
+		
+		
+		addComponent(new JLabel("Way Point File"),0,.1,.1,.1);
+		JButton wayPointFileSelectBtn = new JButton("Select File");
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/ConfigurationPanel.java]: Codacy Issue found: [Avoid excessively long variable names like wayPointFileSelectBtn](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2264186471/issues/source?bid=2938665&fileBranchId=3180106#l86)

**Comment by VivaanBahl** [comment]
I am signing off and merging

**Pull request merged**